esp/services/ws_smc/ws_smcService.cpp

```diff
@@ -2344,3 +2345,181 @@ void CWsSMCEx::setActiveWUs(IEspContext &context, IEspActiveWorkunit& wu, IEspAc
+    timeLocked.getString(timeStr);
+    lock->setTimeLocked(timeStr.str());
+
+    unsigned mode = lD.mode;
+    VStringBuffer modeStr("%x", mode);
+    lock->setModes(modeStr.str());
+
+    StringArray modes;
+    if (RTM_MODE(mode, RTM_LOCK_READ))
+        modes.append(LockModeNames[CLockModes_READ]);
+    if (RTM_MODE(mode, RTM_LOCK_WRITE))
+        modes.append(LockModeNames[CLockModes_WRITE]);
+    if (RTM_MODE(mode, RTM_LOCK_HOLD)) // long-term lock
+        modes.append(LockModeNames[CLockModes_HOLD]);
+    if (RTM_MODE(mode, RTM_LOCK_SUB)) // locks all descendants as well as self
+        modes.append(LockModeNames[CLockModes_SUB]);
+    lock->setModeNames(modes);
+    locks.append(*lock.getClear());
+}
+
+bool CWsSMCEx::onLockQuery(IEspContext &context, IEspLockQueryRequest &req, IEspLockQueryResponse &resp)
+{
+    class CLockPostFilter
+    {
+        CLockModes mode;
+        time_t ttLTLow, ttLTHigh;
+        bool checkLTLow, checkLTHigh;
+        int durationLow, durationHigh;
+
+        bool checkMode(unsigned lockMode)
+        {
+            unsigned modeReq;
+            switch (mode)
+            {
+            case CLockModes_READ:
+                modeReq = RTM_LOCK_READ;
+                break;
+            case CLockModes_WRITE:
+                modeReq = RTM_LOCK_WRITE;
+                break;
+            case CLockModes_HOLD:
+                modeReq = RTM_LOCK_HOLD;
+                break;
+            case CLockModes_SUB:
+                modeReq = RTM_LOCK_SUB;
+                break;
+            default:
+                return true;
+            }
+            if (lockMode & modeReq)
+                return true;
+
+            return false;
+        }
+    public:
+        CLockPostFilter(IEspLockQueryRequest& req)
+        {
+            mode = req.getMode();
+            if (mode == LockModes_Undefined)
+                throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid Lock Mode");
+
+            if (req.getDurationMSLow_isNull())
+                durationLow = -1;
+            else
+                durationLow = req.getDurationMSLow();
+            if (req.getDurationMSHigh_isNull())
+                durationHigh = -1;
+            else
+                durationHigh = req.getDurationMSHigh();
+            const char* timeLow = req.getTimeLockedLow();
+            if (!timeLow || !*timeLow)
+                checkLTLow = false;
+            else
+            {
+                CDateTime dtLow;
+                dtLow.setString(timeLow, NULL, false);
+                ttLTLow = dtLow.getSimple();
+                checkLTLow = true;
+            }
+            const char* timeHigh = req.getTimeLockedHigh();
+            if (!timeHigh || !*timeHigh)
+                checkLTHigh = false;
+            else
+            {
+                CDateTime dtHigh;
+                dtHigh.setString(timeHigh, NULL, false);
+                ttLTHigh = dtHigh.getSimple();
+                checkLTHigh = true;
+            }
```

Comment:
@wangkx - noticed (during a build) that the above code can leave ttLTHigh and ttLTLow uninitialized.
Resulting in compile warnings:
```
/home/jsmith/git/HPCC-Platform/esp/services/ws_smc/ws_smcService.cpp: In member function ‘virtual bool CWsSMCEx::onLockQuery(IEspContext&, IEspLockQueryRequest&, IEspLockQueryResponse&)’:
/home/jsmith/git/HPCC-Platform/esp/services/ws_smc/ws_smcService.cpp:2474:29: warning: ‘postFilter.CWsSMCEx::onLockQuery(IEspContext&, IEspLockQueryRequest&, IEspLockQueryResponse&)::CLockPostFilter::ttLTHigh’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             if (checkLTHigh && (ttNow - duration/1000 > ttLTHigh))
                             ^
/home/jsmith/git/HPCC-Platform/esp/services/ws_smc/ws_smcService.cpp:2472:28: warning: ‘postFilter.CWsSMCEx::onLockQuery(IEspContext&, IEspLockQueryRequest&, IEspLockQueryResponse&)::CLockPostFilter::ttLTLow’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             if (checkLTLow && (ttNow - duration/1000 < ttLTLow))
                            ^

```

Shouldn't have any adverse effect, but worth fixing.